### ISSUE TITLE
How can I run this on my pc?

### ISSUE THREAD

**Issue by andystevens94** [AmarSaini/Clash-Royale-AI-Card-Tracker#1]
I have downloaded the zip, how can I run the script?
Thanks

**Comment by AmarSaini** [comment]
Have Bluestacks running.
Maximize blue stack window while keeping it in windowed mode.
Make sure you have 1920x1080 resolution, as the images are cropped by hard coded pixels.
Start the script. And wait for it to ask you for the starting elixir.
Start a game on your phone, then on bluestacks, spectate that live game.
Enter the starting elixir.

**Comment by andystevens94** [comment]
I have installed numpy, keras and tensorflow but get these error any ideas what I can do?

![pythonerrors](https://user-images.githubusercontent.com/26678082/45252595-0573d400-b351-11e8-96ec-b8f7deb4de06.png)

**Comment by AmarSaini** [comment]
Make sure all of those packages are updated. Try using "pip install" for numpy, keras, and tf

**Comment by eiby1993** [comment]
What scaling size % , did you use for your screen display? Bluestack version and aspect ratio + DPI setting? 
The app runs on my end , however the count elixir goes negative and card images start to shuffle after fifth card is revealed. I am using 1920x1080 resolution. I did noticed that the 8 images are being cropped not perfectly and that's affecting the percentage% on the script.
I appreciate your incredible work.

**Comment by AmarSaini** [comment]
Hey!

I believe I used a scaling size of 125%,
1920 x 1080 is definitely needed.
As for blue stacks, try just maximizing it in WINDOW mode.

Attach some of those cropped images!

**Comment by eiby1993** [comment]
I have Bluestacks maximized on window mode... It looks like the top bar of Bluestacks is taking more space than it should and it's pushing down the region size of the game. I'd take any advices at this point and hope these pics of cropped images (at the bottom of each) helps.  
Note: I believe the percentage for each card on CNN1 should be 99.99% , it shows 50% when images are cropped incorrectly. I am using this an indicator when troubleshooting.
![cap1](https://user-images.githubusercontent.com/22260595/48656040-b18b0880-e9ed-11e8-9458-27b50e87c406.GIF)
![cap2](https://user-images.githubusercontent.com/22260595/48656041-b18b0880-e9ed-11e8-9fba-2495d528d2c7.GIF)

**Comment by AmarSaini** [comment]
Hmmmm I see what you mean!

By any chance do you know some programming?
I can direct you to the area of the code, and we can try changing the y-value for the crop (perhaps shift it down).

I'll see what else I can find.
Also, what scaling are you on? Maybe try 100%

**Comment by eiby1993** [comment]
Yes !! I know some python.  I'd appreciate if you can point me to the area of code ( line of code #). Last winter, I was learning about PIL library however i did a project nothing near yours.
I tried using 100%, 110% , 125% and near values between those numbers ...  None of them give me a solid and consistent behavior. Most of them, start to shuffle cards after fifth card is revealed and elixir count goes negative lol.

**Comment by AmarSaini** [comment]
Sure I'll let you know what line asap,

Also if you want to debug more, check the "cropped" folders, they should hold individual images that are being cropped at every iteration.

Also, what kind of specs do you have on your cpu? I ask this because neural networks take some time to run, and you may want to have a somewhat fast CPU to keep up with each iteration of the images.

**Comment by eiby1993** [comment]
Thank you so much ! I've been checking those "testdata" folders .
I have a core i7 6700HQ 2.60Ghz  and 16GB of RAM...  Hope that's enough lol

**Comment by AmarSaini** [comment]
Should be more than enough

**Comment by AmarSaini** [comment]
These test images are cropped for 2 different networks. (testData and testData2)

The first, testData, are for the actual cards. These go into the first neural network.

The second, testData2, are for the area above the cards. These are for detecting when cards are played. These go into the second neural network.

More info can be found in the pdf explaining the project.

As for the lines of code.
The functions loadTestImages1() and loadTestImages2() are in charge of cropping the right areas of your screen. A screenshot of your screen is taken before these function, and saved as "testCNN.png".
This all occurs in the while(True) loop of the main function: liveBothModelPredicts().

Anyway, see the attached images for where the images are cropped inside these 2 loadTestImage() functions.

The format arr[y1:y2, x1:x2] basically means what range of y and x values to crop from. Yes the y values are first

You would want to play with the y1:y2, or in this case, the "58:169" in loadTestingImages1(). And maybe the "58:88" in loadTestingImages2()

**Comment by AmarSaini** [comment]
Let me know if you have any questions in regards to that last email.

Be sure to check out that report to see some examples of how the images are formatted

**Comment by eiby1993** [comment]
Thank you so much ! I'll play with those numbers and let you know how it goes.. Have a good day !
Note: I did read the report... it was a great document to use a reference. 👍

**Comment by AmarSaini** [comment]
No problem!

**Comment by AmarSaini** [comment]
Hey also,

This may matter, what kind of battles are you playing? Friendly, ladder, challenges?

**Comment by eiby1993** [comment]
friendly :S

**Comment by AmarSaini** [comment]
Friendly battles show your opponent's elixir count. There'll be an elixir bar above the opponent's cards in spectator mode, causing the cards to be shifted down a little bit. This may affect the cropping
!

**Comment by eiby1993** [comment]
I didn’t pay attention to that detail . Thanks . It runs fine know ! ;)

**Comment by AmarSaini** [comment]
Alrighty!

**Comment by mrtrick3737** [comment]
Just a note on MacOS, had to change any of the tkinter lines to start with a capital Tk to get it to run.

**Comment by AmarSaini** [comment]
Check this out

https://stackoverflow.com/questions/27563854/why-cant-the-import-command-be-found

________________________________

**Comment by AmarSaini** [comment]
Check this out

https://stackoverflow.com/questions/27563854/why-cant-the-import-command-be-found

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by mrtrick3737** [comment]
Is this expecting a particular version of python? I get it to run in 2.7 but not correctly yet. Can't get it to run in 3.x

**Comment by AmarSaini** [comment]
Definitely Python 3

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by A-Koutiney** [comment]
hey, i get this error message when i run the script:
" import cv2
ModuleNotFoundError: No module named 'cv2' "

**Comment by AmarSaini** [comment]
Looks like you need to install the library for cv2, check out how to use pip to install it.

Amar Saini

B.S. - Computer Science & Engineering
School of Engineering, University of California, Merced

________________________________

**Comment by A-Koutiney** [comment]
i did installed  opencv and whenever i typed import cv2 in python environment, it says No module named 'cv2'

**Comment by AmarSaini** [comment]
My guess it hat you used "pip install opencv-python" would I believe is for Python 2.7, try using "pip3 install opencv-python"

Amar Saini

B.S. - Computer Science & Engineering
School of Engineering, University of California, Merced

________________________________
